lib/index.js

```diff
@@ -457,6 +458,15 @@ function nop() {
 function pipeReadable(xs, stream) {
     // write any errors into the stream
     xs.on('error', writeStreamError);
+
+    // stream is the thing being returned, and shouldn't be writable
+    // but pipe needs something writable
+    proxy = function () {
```

Comment:
Right, right, good catch. Implementing the proxy correctly ended up being a pain so I switched strategies a bit.